api/src/main/java/org/openmrs/module/pihcore/setup/LocationTagSetup.java

```diff
@@ -16,17 +16,66 @@
 import org.openmrs.module.pihcore.metadata.liberia.LiberiaLocations;
 import org.openmrs.module.pihcore.metadata.mexico.MexicoLocations;
 import org.openmrs.module.pihcore.metadata.sierraLeone.SierraLeoneLocations;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.lang.reflect.Field;
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.List;
 
 public class LocationTagSetup {
 
-    // TODO make this less if-then based
-    // TODO actually, when we get rid of these, we can probably set most of the tags in the bundles?
+    private static final Logger log = LoggerFactory.getLogger(LocationTagSetup.class);
+
     public static void setupLocationTags(LocationService locationService, Config config) {
+        for (String locationTag : config.getLocationTags().keySet()) {
+            setLocationTagsFor(locationService, locationTag, config.getLocationTags().get(locationTag));
+        }
+    }
+
+    private static void setLocationTagsFor(LocationService service, String locationTag, List<String> locationsThatGetTag) {
```

Comment:
I took the existing setLocationTagsFor method and changed it to operate on a String instead of a LocationTagDescriptor and  a List<String> instead of a List<LocationDescriptor>.  I also moved the method up to the top of the class, which may not have been the best thing in terms of reviewing because you can't see the proper diff.